packages/xo-web/src/xo-app/backup/health/index.js

```diff
@@ -1,17 +1,87 @@
 import _ from 'intl'
-import Component from 'base-component'
+import ActionButton from 'action-button'
+import decorate from 'apply-decorators'
 import Icon from 'icon'
 import Link from 'link'
 import NoObjects from 'no-objects'
 import React from 'react'
-import renderXoItem from 'render-xo-item'
+import renderXoItem, { BackupJob, Vm } from 'render-xo-item'
 import SortedTable from 'sorted-table'
-import { addSubscriptions, connectStore } from 'utils'
+import { addSubscriptions, connectStore, noop } from 'utils'
 import { Card, CardHeader, CardBlock } from 'card'
 import { Container, Row, Col } from 'grid'
+import { confirm } from 'modal'
+import { createPredicate } from 'value-matcher'
 import { createGetLoneSnapshots, createGetObjectsOfType } from 'selectors'
-import { deleteSnapshot, deleteSnapshots, subscribeSchedules } from 'xo'
-import { FormattedRelative, FormattedTime } from 'react-intl'
+import { forEach, keyBy, omit, toArray } from 'lodash'
+import { FormattedDate, FormattedRelative, FormattedTime } from 'react-intl'
+import { get } from '@xen-orchestra/defined'
+import { injectState, provideState } from 'reaclette'
+import {
+  deleteBackups,
+  deleteSnapshot,
+  deleteSnapshots,
+  getRemotes,
+  listVmBackups,
+  subscribeBackupNgJobs,
+  subscribeSchedules,
+} from 'xo'
+
+const DETACHED_BACKUP_COLUMNS = [
+  {
+    name: _('date'),
+    itemRenderer: backup => (
+      <FormattedDate
+        value={new Date(backup.timestamp)}
+        month='long'
+        day='numeric'
+        year='numeric'
+        hour='2-digit'
+        minute='2-digit'
+        second='2-digit'
+      />
+    ),
+    sortCriteria: 'timestamp',
+    sortOrder: 'desc',
+  },
+  {
+    name: _('vm'),
+    itemRenderer: ({ vm, vmId }) => <Vm id={vmId} link name={vm.name_label} />,
+    sortCriteria: ({ vmId }, { vms }) => get(() => vms[vmId].name_label),
```

Comment:
You could fallback to the name saved in the record, in case the VM doesn't exist.